packages/dev/core/src/Lights/rectAreaLight.ts

```diff
@@ -0,0 +1,194 @@
+import { Vector3 } from "../Maths/math.vector";
+import { Node } from "../node";
+import { Light } from "./light";
+import type { Effect } from "core/Materials/effect";
+import { RegisterClass } from "core/Misc/typeStore";
+import { DefaultAreaLightLTCProvider } from "core/Lights/LTC/ltcTextureTool";
+import type { IAreaLightLTCProvider } from "core/Lights/LTC/ltcTextureTool";
+import { serialize } from "../Misc/decorators";
+import type { Scene } from "core/scene";
+
+Node.AddNodeConstructor("Light_Type_4", (name, scene) => {
+    return () => new RectAreaLight(name, Vector3.Zero(), 1, 1, scene);
+});
+
+declare module "../scene" {
+    export interface Scene {
+        areaLightLTCProvider: IAreaLightLTCProvider;
+    }
+}
+
+function IsAreaLightsReady(scene: Scene): boolean {
+    if (scene.areaLightLTCProvider.ltc1Texture && scene.areaLightLTCProvider.ltc2Texture) {
+        return scene.areaLightLTCProvider.ltc1Texture.isReady() && scene.areaLightLTCProvider.ltc2Texture.isReady();
+    }
+
+    return false;
+}
+
+/**
+ * A rectangular area light defined by an unique point in world space, a width and a height.
+ * The light is emitted from the rectangular area in the -Z direction.
+ */
+export class RectAreaLight extends Light {
```

Comment:
Do we want to already introduce AreaLight to be the parent of REct and circle later? to avoid breaking backward compat?

Reply:
My thinking is that future Area Lights could inherit from RectAreaLight.